.gitmodules

```diff
@@ -262,6 +262,9 @@
 [submodule "contrib/minizip-ng"]
 	path = contrib/minizip-ng
 	url = https://github.com/zlib-ng/minizip-ng
+[submodule "contrib/spotify-annoy"]
+	path = contrib/spotify-annoy
+	url = https://github.com/Vector-Similarity-Search-for-ClickHouse/annoy.git
```

Comment:
I can't find but we had error with sanitiser that was fixed in this commit https://github.com/Vector-Similarity-Search-for-ClickHouse/annoy/commit/301ff04e2213abaa7cbe30041b9b576c968bd994

Reply:
https://s3.amazonaws.com/clickhouse-builds/37215/bdb66914051da3376fd2d65ecc93cad757ab22ce/binary_tidy/build_log.log -- failed test

Reply:
The warning was actually generated by clang-tidy and not the "true" sanitizers (asan, ubsan, tsan etc.). In theory, clang-tidy checks only *source files* in directories src, utils, programs and base. So I am a bit surprised that the warning is generated at all.
Perhaps it would be a better solution to draft a PR to the original Annoy repo which adds a `// NOLINT(clang-analyzer-unix.Malloc)` to the statement instead of maintaining our own fork?

Reply:
EDIT: I learned today, that there are two defined ways to integrate a *patched* version of an external library into ClickHouse:
1. upstream the patch to the original lib (like I proposed above) and reference in .gitmodules the upstream repository. This is the preferred way as it removes the maintenance burden for the patch from the ClickHouse team.
2.  create a fork of the original library under the ClickHouse umbrella (see the many forks of external libs in https://github.com/orgs/ClickHouse/repositories), patch the fork and reference in .gitmodules the forked repository. A bit more work but also okay.